### PR TITLE
Feat: add refresh life cycles

### PR DESCRIPTION
**Pull request by yy-wow**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

### What is the new behavior?
1. 修复源码调试找不到包问题；
2. 刷新工具栏增加生命周期方法;
3. 增加画布标识，控制是否允许拖拽、选中组件、允许原生事件触发等

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Release Notes

- **New Features**
  - Introduced a design mode feature, allowing users to switch between design and runtime states for enhanced control over canvas behavior.
  - Added notification handling and improved refresh logic in the main component for better user feedback during canvas operations.

- **Bug Fixes**
  - Corrected a typo in the refresh function to ensure proper functionality.

- **Improvements**
  - Enhanced event handling by conditionally executing based on the current design mode, improving responsiveness and user experience.

- **Documentation**
  - Updated constants and functions related to design mode for clarity and usability.

### CODE REVIEW

File: packages/canvas/render/src/context.js [hunk 1]
```diff
@@ -55,3 +55,12 @@ export const setCondition = (id, visible = false) => {
 export const getCondition = (id) => conditions[id] !== false
 
 export const getConditions = () => conditions
+
+// 是否表现画布内特征的标志，用来控制是否允许拖拽、原生事件是否触发等
+let canvasFlag = true
```
Comment (chilingling) [inline packages/canvas/render/src/context.js]: 建议语义化一点。你这个配置实际上是变成了 `designMode`  和 `previewMode` 了

File: packages/canvas/container/src/CanvasContainer.vue [hunk 2]
```diff
@@ -168,7 +180,9 @@ export default {
         })
 
         win.addEventListener('mousemove', (ev) => {
-          dragMove(ev, true)
+          isCanvasEvent(ev, (e) => {
```
Comment (chilingling) [inline packages/canvas/container/src/CanvasContainer.vue]: 这里是不是都可以直接不监听？不需要再监听回调函数里面处理？

File: packages/canvas/render/src/context.js (hunk 1 above)
Comment (hexqi) [inline packages/canvas/render/src/context.js]: 可以考虑换个名字，当前看不出来作用

File: packages/canvas/container/src/CanvasContainer.vue [hunk 3]
```diff
@@ -117,6 +117,16 @@ export default {
       }
     }
 
+    const isCanvasEvent = (event, handler) => {
```
Comment (hexqi) [inline packages/canvas/container/src/CanvasContainer.vue]: is开头方法主要用来判断条件返回true/false，传入回调方法感觉很奇怪

File: packages/canvas/container/src/CanvasContainer.vue [hunk 4]
```diff
@@ -117,6 +117,16 @@ export default {
...
+      const canvasFlag = canvasApi.getRenderer().getCanvasFlag()
```
Comment (chilingling) [inline packages/canvas/container/src/CanvasContainer.vue]: 这里为什么要读取 canvas 里面的配置？应该是 canvas 画布依赖外部的配置。外部不要在加对 canvas 画布的依赖？

File: packages/canvas/render/src/context.js (hunk 1 above)
Comment (yy-wow) [reply packages/canvas/render/src/context.js]: 更换为isDesignMode、getDesignMode、setDesignMode

File: packages/canvas/container/src/CanvasContainer.vue (hunk 3 above)
Comment (yy-wow) [reply packages/canvas/container/src/CanvasContainer.vue]: 更名为handleCanvasEvent

File: packages/canvas/render/src/context.js (hunk 1 above)
Comment (yy-wow) [reply packages/canvas/render/src/context.js]: 更换为isDesignMode、getDesignMode、setDesignMode

File: packages/canvas/container/src/CanvasContainer.vue (hunk 2 above)
Comment (yy-wow) [reply packages/canvas/container/src/CanvasContainer.vue]: 需要支持切换，不能直接不监听

File: packages/canvas/container/src/CanvasContainer.vue (hunk 4 above)
Comment (yy-wow) [reply packages/canvas/container/src/CanvasContainer.vue]: 加这个标识有两个方面：1.unibuy有个切换画布状态的，是否允许画布内组件的交互事件；2.运行时渲染，区分是运行时还是编辑态。
所以这个配置还是需要画布自身维护

File: packages/canvas/container/src/CanvasContainer.vue [hunk 5]
```diff
@@ -131,15 +141,17 @@ export default {
...
-          // html元素使用scroll和mouseup事件处理
-          if (event.target === doc.documentElement) {
-            isScrolling = false
-            return
-          }
-
-          insertPosition.value = false
-          setCurrentNode(event)
-          target.value = event.target
+          handleCanvasEvent(event, () => {
```
Comment (gene9831) [inline packages/canvas/container/src/CanvasContainer.vue]: `handleCanvasEvent` 的第二个参数，这里传了一个没有参数的函数，但是在127行调用时，传入了 `event` 参数,，实际上没用到 `event` 参数。在 `handleCanvasEvent` 函内部，`event` 参数的唯一作用是传给`handler`，所以 `event` 完全可以不传入，在传入 `handler` 时生成闭包可以获取 `event`

File: packages/toolbars/refresh/src/Main.vue [hunk 6]
```diff
@@ -48,10 +69,29 @@ export default {
...
+      try {
+        if (typeof beforeRefresh === 'function') {
+          await beforeRefresh()
+        }
+
+        if (typeof refreshMethod === 'function') {
+          const stop = await refreshMethod()
+
+          if (stop) {
+            return
```
Comment (gene9831) [inline packages/toolbars/refresh/src/Main.vue:74]: 这里的 `return` 似乎没什么意义，是不是少了 `return` 的值

File: packages/canvas/render/src/render.js [hunk 7]
```diff
@@ -558,16 +558,20 @@ const getBindProps = (schema, scope) => {
   const bindProps = {
     ...parseData(schema.props, scope),
     [DESIGN_UIDKEY]: id,
-    [DESIGN_TAGKEY]: componentName,
-    onMouseover: stopEvent,
-    onFocus: stopEvent
+    [DESIGN_TAGKEY]: componentName
   }
+
+  if (getDesignMode()) {
```
Comment (hexqi) [inline packages/canvas/render/src/render.js]: `if (getDesignMode())` 建议改成 `if (getCanvasMode() === CANVAS_MODE.Design)`, 意思能够更明确（增加CANVAS_MODE常量、canvasMode变量）

File: packages/canvas/container/src/CanvasContainer.vue (hunk 5 above)
Comment (yy-wow) [reply packages/canvas/container/src/CanvasContainer.vue]: 已修改

File: packages/toolbars/refresh/src/Main.vue (hunk 6 above)
Comment (yy-wow) [reply packages/toolbars/refresh/src/Main.vue:74]: 这里return是跳出当前函数执行，这里的含义是传入一个标识，用来判断后面的逻辑是否继续执行

File: packages/canvas/render/src/render.js (hunk 7 above)
Comment (yy-wow) [reply packages/canvas/render/src/render.js]: 已修改

**Review by hexqi — APPROVED** [review]

**Review by chilingling — APPROVED** [review]

**Pull request merged**